utils/s2n_init.c

```diff
@@ -42,18 +38,14 @@ int s2n_init(void)
 }
 
 int s2n_cleanup(void)
-{
-    GUARD(s2n_rand_cleanup_thread());
-
-    return 0;
-}
-
-static void s2n_cleanup_atexit(void)
 {
     s2n_rand_cleanup_thread();
     s2n_cipher_suites_cleanup();
```

Comment:
There is an issue with moving these functions to `s2n_cleanup`. `s2n_cleanup` is supposed to be called before any thread exists. With this change, if *any* thread exits, some global state will be wiped, causing all other threads to no longer function properly. This is an issue for any application that spins up a thread per connection.